pkg/utils/jsonutils.go

```diff
@@ -58,3 +63,61 @@ func FlattenSingleValue(key string, m map[string]interface{}, child interface{})
 		m[key] = child
 	}
 }
+
+func ExtractJsonValueBuffered(data []byte, workBuf []byte, keys ...string) ([]byte, jsonparser.ValueType, error) {
+	var err error
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	decoder.UseNumber()
+	var current interface{} = nil
+	for _, key := range keys {
+		if err = decoder.Decode(&current); err != nil {
+			return nil, jsonparser.NotExist, fmt.Errorf("error decoding JSON: %v", err)
+		}
+		switch obj := current.(type) {
+		case map[string]interface{}:
+			if value, found := obj[key]; found {
+				current = value
+			} else {
+				return nil, jsonparser.NotExist, fmt.Errorf("key %s not found", key)
+			}
+		case []interface{}:
+			return nil, jsonparser.NotExist, fmt.Errorf("unexpected array while traversing key %s", key)
+		default:
+			return nil, jsonparser.NotExist, fmt.Errorf("expected object for key %s but found %T", key, obj)
+		}
+	}
+	valueBytes, err := json.Marshal(current)
+	if err != nil {
+		return nil, jsonparser.NotExist, fmt.Errorf("error converting value to bytes: %v", err)
+	}
+	workBuf = ResizeSlice(workBuf, len(valueBytes))
+	copy(workBuf, valueBytes)
+	var valueType jsonparser.ValueType
+	switch current.(type) {
+	case string:
+		valueType = jsonparser.String
+	case json.Number:
+		valueType = jsonparser.Number
+	case bool:
+		valueType = jsonparser.Boolean
+	case nil:
+		valueType = jsonparser.Null
+	default:
+		valueType = jsonparser.Object
+	}
+	return workBuf, valueType, nil
+}
+func GetStringFromJson(data []byte, workBuf []byte, keys ...string) ([]byte, error) {
+	workBuf, dataType, err := ExtractJsonValueBuffered(data, workBuf, keys...)
+	if err != nil {
+		return nil, err
+	}
+	if dataType != jsonparser.String {
+		if dataType == jsonparser.Null {
+			return nil, fmt.Errorf("key %s has a null value", strings.Join(keys, ", "))
+		}
+		return nil, fmt.Errorf("expected string value for key(s) %s but got %s", strings.Join(keys, ", "), dataType)
+	}
+
+	return []byte(strings.Trim(string(workBuf), "\"")), nil
```

Comment:
There seems to be a misunderstanding. The entire point of this PR (to fix https://github.com/siglens/siglens/issues/1538) is to reduce memory usage. Here you're doing something like `[]byte(string(workBuf))`, which is going to use more memory